src/Compilers/CSharp/Test/Emit2/Attributes/AttributeTests_WellKnownAttributes.cs

```diff
@@ -13803,5 +13803,44 @@ public struct S2 { }
                 Diagnostic(ErrorCode.ERR_InvalidNamedArgument, "CharSet=0").WithArguments("CharSet").WithLocation(7, 38)
                 );
         }
+
+        [Fact, WorkItem(64605, "https://github.com/dotnet/roslyn/issues/64605")]
+        public void ObsoleteWithInterpolation()
+        {
+            var source = """
+using System;
+
+internal class Program
+{
+    static void Main(string[] args)
+    {
+        // this does not raise a warning
+        var object1 = new LegacyObject1();
+
+        // but this does
+        var object2 = new LegacyObject2();
+    }
+}
+
+[Obsolete($"Do not use {nameof(LegacyObject1)}")]
```

Comment:
Let's add some dedicated tests for when the interpolated string has non-constant underlying components in this location.